server/planning/item_lock.py

```diff
@@ -0,0 +1,149 @@
+# -*- coding: utf-8; -*-
+#
+# This file is part of Superdesk.
+#
+# Copyright 2013, 2014 Sourcefabric z.u. and contributors.
+#
+# For the full copyright and license information, please see the
+# AUTHORS and LICENSE files distributed with this source code, or
+# at https://www.sourcefabric.org/superdesk/license
+
+import logging
+import superdesk
+
+from superdesk.errors import SuperdeskApiError
+from superdesk.notification import push_notification
+from superdesk.users.services import current_user_has_privilege
+from superdesk.utc import utcnow
+from superdesk.lock import lock, unlock
+from eve.utils import config
+from superdesk import get_resource_service
+from apps.common.components.base_component import BaseComponent
+
+
+LOCK_USER = 'lock_user'
+LOCK_SESSION = 'lock_session'
+logger = logging.getLogger(__name__)
+
+
+class ItemLockPlanningModule(BaseComponent):
+    def __init__(self, app):
+        self.app = app
+        self.app.on_session_end += self.on_session_end
+
+    @classmethod
+    def name(cls):
+        return 'item_lock_planning'
+
+    def lock(self, item_id, user_id, session_id, action, resource):
+        item_service = get_resource_service(resource)
+        item = item_service.find_one(req=None, _id=item_id)
+
+        # set the lock_id it per item
+        lock_id = "item_lock {}".format(item.get(config.ID_FIELD))
+
+        if not item:
+            raise SuperdeskApiError.notFoundError()
```

Comment:
`if not item` should be above the `lock_id =` assignment
as `.format(item.get` will fail if item is None